activiti-core/activiti-spring/src/main/java/org/activiti/spring/autodeployment/AbstractAutoDeploymentStrategy.java

```diff
@@ -128,13 +128,21 @@ private boolean isProcessDefinitionResource(String resource) {
     }
 
     protected DeploymentBuilder loadProjectManifest(DeploymentBuilder deploymentBuilder) {
-        if (projectModelService != null && projectModelService.hasProjectManifest()) {
+        if (applicationUpgradeContextService != null && applicationUpgradeContextService.hasProjectManifest()) {
             try {
-                deploymentBuilder.setProjectManifest(projectModelService.loadProjectManifest());
+                deploymentBuilder.setProjectManifest(applicationUpgradeContextService.loadProjectManifest());
             } catch (IOException e) {
                 LOGGER.warn("Manifest of application not found. Project release version will not be set for deployment.");
             }
         }
+
+        if(applicationUpgradeContextService != null && applicationUpgradeContextService.hasEnforcedAppVersion()){
```

Comment:
Will `applicationUpgradeContextService` ever be null? I don't think so, and that should be some sort of invalid state of the application, and checked earlier (constructor)

Reply:
Yes, applicationUpgradeContextService can actually be null, since its instantiation depends on the existence of a manifest file in the project, something that does not happing in the engine modules.
But can perform a null check before this point, sure 👍